src/middlewared/middlewared/plugins/disk_/encryption_freebsd.py

```diff
@@ -75,17 +75,9 @@ def geli_attach_single(self, dev, key, passphrase=None, skip_existing=False):
             )
             if cp.stderr or not os.path.exists(f'/dev/{dev}.eli'):
                 raise CallError(f'Unable to geli attach {dev}: {cp.stderr.decode()}')
-            self.__geli_notify_passphrase(passphrase)
         else:
             self.logger.debug(f'{dev} already attached')
 
-    def __geli_notify_passphrase(self, passphrase):
-        if passphrase:
-            with open(passphrase) as f:
-                self.middleware.call_hook_sync('disk.post_geli_passphrase', f.read())
-        else:
-            self.middleware.call_hook_sync('disk.post_geli_passphrase', None)
```

Comment:
Was this not used?

Reply:
We were using this hook before but with this hook we are not able to map a disk with a pool. So I changed this and we are using import/export/unlock/lock/passphrase hooks to keep the cache updated.